app/services/course/assessment/question/koditsu_question_service.rb

```diff
@@ -78,11 +78,70 @@ def build_question_object
         templates: [{
           path: koditsu_programming_language_map[@type][:filename],
           content: @metadata['submission'],
-          prefix: @metadata['prepend'],
-          suffix: koditsu_programming_language_map[@type][:language] == 'cpp' ? '' : @metadata['append']
+          prefix: truncate_google_test_framework_and_clean_comments(@metadata['prepend']),
+          suffix: truncate_google_test_framework_and_clean_comments(@metadata['append'])
         }],
         exprTestcases: @test_cases
       }]
     }
   end
+
+  def clean_from_comments(snippet)
+    no_single_line_comments_snippet = snippet.gsub(/\/\/.*$/, '')
+    no_multiple_line_comments_snippet = no_single_line_comments_snippet.gsub(/\/\*.*?\*\//m, '')
+
+    no_multiple_backspaces = no_multiple_line_comments_snippet.
+                             gsub(/\r/, '').
+                             gsub(/\n\s{4}/, "\n").
+                             gsub(/\n{2,}/, "\n")
+
+    no_multiple_backspaces.strip
```

Comment:
No need to strip, since this function only intends to clean comments.

Reply:
done